tests/unit/package_managers/bundler/test_parser.py

```diff
@@ -180,3 +181,25 @@ def test_parse_gemlock_empty(
 
     assert f"Package {rooted_tmp_path.path.name} is bundled with version 2.5.10" in caplog.messages
     assert result == []
+
+
+@pytest.mark.parametrize(
+    "source",
+    [
+        "https://rubygems.org",
+        "https://dedicatedprivategemrepo.com",
+    ],
+)
+@mock.patch("cachi2.core.package_managers.bundler.parser.download_binary_file")
+def test_dependnecies_could_be_downloaded(
+    mock_downloader: mock.MagicMock,
+    source: str,
+) -> None:
+    base_destination = Path("/tmp/foo")
```

Comment:
You can use `rooted_tmp_path` fixture instead.

Reply:
I could, but this test does not touch file system at all, it only verifies the flow. There is no need to use a real path, anything path-like would do.

Reply:
Which makes me wonder if we should try to keep the unit tests leaner.  We have already set a lot of precedent by having many tests that actually reach out for the file system, but I'm not sure if the added execution time and complexity are worth it (for example, [this fixture](https://github.com/containerbuildsystem/cachi2/blob/6a483062359b0265cd208861f74a6f35b3e37f07/tests/unit/package_managers/conftest.py#L188) in another bundler test that generates a full git repo). 

I probably still tend to see unit tests as very isolated, so I might be biased here.